### ISSUE TITLE
addOverrideParam does funny things

### ISSUE THREAD

**Issue by tobystokes** [vaersaagod/geomate#44]
Hi,
Since 1.3.1, using ` | addOverrideParam` behaves a bit unexpectedly:

```
{{ '/anyold_string/' | addOverrideParam }}  -> /anyold_string/?__region=✪
```
But on https://domain.com/site/path
```
{{ craft.app.request.pathInfo | addOverrideParam }} -> https://domain.com/site/https://domain.com/site/path?__region=✪
```
Whereas previously it returned 
```
{{ craft.app.request.pathInfo | addOverrideParam }} -> path?__region=✪
```
given that 
```
{{ craft.app.request.pathInfo }} -> path
```

For context, I'm using pathInfo in the navigation rather than `craft.geomate.getSiteLinks()` so i can send visitors directly to the same page on the alternate site by just replacing the site slug, but if there is a better way to do that, please let me know!

**Comment by mmikkel** [comment]
Hi @tobystokes, this behaviour changed a bit in 1.3.1 due to the `addOverrideParam` filter now using [`UrlHelper::url()`](https://github.com/vaersaagod/geomate#addoverrideparam) under the hood (among other reasons, in order to fix #42). To that end, the filter now works similarly to using Craft's built-in `url()` method, which means that it'll prepend the site URL to the value if the value doesn't represent an absolute or root-relative URL.

Whilst this is technically a breaking change, I'm hesitant to reintroduce the old behaviour as it doesn't make sense to add the override param to something that isn't a URL, and I'd consider the old behaviour a bug. [The documentation](https://github.com/vaersaagod/geomate#addoverrideparam) is also clear that the value should be a URL.

To fix the issue on your end, I suppose the easiest way would be to make the path a root-relative URL, i.e. `{{ ('/' ~ craft.app.request.pathInfo) | addOverrideParam) }}`.

**Comment by tobystokes** [comment]
Aha, fair enough, I had my `/` before the twig tag, but concatenating like that works fine and makes more sense.
Thanks for this excellent plugin!
